### ISSUE TITLE
TODO issue points to wrong section of code

### ISSUE THREAD

**Issue by hawkins** [JasonEtco/todo#131]
Hey again! 👋 

I opened a [PR](https://github.com/hawkins/resume/pull/8) on one of my repos earlier and added a TODO comment, but the issue that was reported from this bot was pointing to the wrong section of code. I.e., it pointed to lines 136-141 when it should have pointed to the ballpark of 170. 

Here's a [link to the line in the diff where I added the TODO](https://github.com/hawkins/resume/pull/8/commits/1cb30710fc989458aa44b26d07ce07723f810c5b#diff-ff76033152632142d10969982c32eb7eR169).

And the [subsequent issue opened by your bot.](https://github.com/hawkins/resume/issues/9)

---

Not sure what I did to cause this yet, haven't had time to diagnose this yet. Have you seen behavior like this before?

**Comment by JasonEtco** [comment]
Hey @hawkins! Thanks for the report!

> Not sure what I did to cause this

You did nothing wrong - this is a bug in TODO that I've seen before, but haven't had time to solve (and kept forgetting to open an issue for).

TLDR, it occurs when there are two hunks in the same file in a diff. That looks like:

```
diff --git a/README.md b/README.md
index 8f65799..f4be99b 100644
--- a/README.md
+++ b/README.md
@@ ... line numbers ... @@
+ changes
- other changes
@@ ... other line numbers ... @@
+ changes
- other changes
```

I believe that right now its basing the line numbers on the _first_ hunk, rather than the _nearest_ (most relevant) hunk numbers.

Totally open to PRs on this, I won't have time to fix it for a few weeks - if you're a RegEx champ then this is for you 😉

**Comment by hawkins** [comment]
Well you know I'm no champ, you've seen me work with RegEx plenty recently 😉  I'll gladly give this a look though! Thanks for the lightning fast feedback! 👏

**Comment by JasonEtco** [comment]
Thinking about this some more, I think it may require more than just a RegEx change - might be best to split hunks by `@@...@@` lines, then parse those hunks. If modifying the RegEx to grab the nearest hunk numbers is possible that'd be best, but it may not be.

**Comment by hawkins** [comment]
I was thinking it looked like a bit more than a regex change at first glance the other day, too; I'm hoping to take a stab at this sometime tomorrow or this weekend, maybe even while streaming if I can come up with a strategy for tackling it first, since a few more eyes on it could help out. 🤷‍♂️

**Comment by hawkins** [comment]
Sorry for the immense delay on this, life’s been comically in the way of my open source time these last few weeks 🙃😔

No hard feelings if you or anyone else would like to tackle this sooner than I can, @jasonetco. Though if it’s still open next time I have time for OSS, I’ll be back to knock it out!🤞

**Comment by JasonEtco** [comment]
I've seen a lot more stable URL references as of #165 so I'm going to close this for now. If you're still seeing issues feel free to reopen ✌️
